person.rb

```diff
@@ -0,0 +1,65 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method #{__method__}"
+  end
+end
```

Comment:
- Kindly move these to a separate file to follow the convention in ruby for separate, maintaining code organization readability.